plugins-base/XSFeatureSRCreate.py

```diff
@@ -1185,7 +1185,13 @@ def CommitCreate(self, inType, inDeviceConfig, inOtherConfig = None):
                 True # shared
                 )
             )
+            # Wait until the task is completed
```

Comment:
You can pass a task to a `ProgressDialogue` and AFAICT it will wait until completion with visual progress (and the possibility to cancel it). It would be nicer if that could be used. There is an example of this in XSFeaturePoolJoin.py.

There are some other examples of async tasks and polling (XSFeatureSRCommon.py and XSFeatureVMCommon.py). I wonder if it would be possible to share the code somehow.

Reply:
Thanks. I'll try `ProgressDialogue`.